_drafts/javascript.md

```diff
@@ -0,0 +1,106 @@
+
+Now go trough each file and improve the code readability by hand as needed. From my experiences with converting Hoodie from CoffeeScript to JavaScript quite a lot of manual work will be required.
+
+### Convert test files from CoffeeScript to JavaScript
+
+Same as the previous steps only for the tests files this time.
+
+When finished, create a separate branch with with the tests written in JavaScript but the source code still in CoffeeScript to assure integrity.
+
+### Update package.json
+
+Remove all CoffeeScript related dependencies and tooling. Update [engines](https://docs.npmjs.com/files/package.json#engines)
+
+Make sure that Hubot scripts can still be written in CoffeeScript. We want to break as few existing scripts as possible and many of them are written in CoffeeScript. We treat it as legacy support and will drop support for scripts written in CoffeeScript in future.
+
+### Add script for JavaScript linting
+
+The only tool I would like to introduce as part of this proposal is a linting tool which will be run as part of `npm test`.
+
+I suggest [standard](https://standardjs.com/). We’ve been using it in all our projects at Hoodie and Neighbourhoodie since 2+ years and never looked back. It's a zero-configuration JavaScript linter.
+
+### Update documentation
+
+Remove any reference of CoffeeScript from the documentation.
+
+## Backward compatibility
+
+Only breaking change is dropping support of Node 0.10 and 0.12.
+
+## Alternatives considered
+
+- **[CoffeeScript 6.0](https://github.com/coffeescript6/discuss)**
+
+  As stated in the [Background section](https://github.com/coffeescript6/discuss#background), JavaScript caught up with most of the features that only CoffeeScript offered in the past.
+
+  The argument for the clear syntax is valid. But we can enforce that with a JavaScript linter.
+
+  The main reason we decided to convert Hubot to JavaScript is to make it easier to contribute to for a wider range of people.
+
+- **Continue to support Node <v4**
+
+  Node versions smaller than v4 are no longer maintained and have known security vulnerabilities. The only major operating system which distributes a no longer maintained Node version by default is Debian, and there are alternative ways to install a recent Node version, see [Debian and Ubuntu based Linux distributions](https://nodejs.org/en/download/package-manager/#debian-and-ubuntu-based-linux-distributions).
+
+  Mikeal Rogers (Node Foundation) says about Debian’s official Node packages:
+
+  > The "official" repositories are outdated and even insecure. We need to treat them as legacy and move on.
+
+  The Node Foundation has tried to get Debian to adapt modern Node versions like "[pretty much every other repository for packages takes modern versions.](https://twitter.com/mikeal/status/869646796888330240)" but could not succeed.
+
+  See more responses to my question regarding old Node versions at https://twitter.com/gr2m/status/869305267464306689
```

Comment:
Debian users that need modern node version have 2 main easy options: 

- turning to [nvm](https://github.com/creationix/nvm)
- getting packages from [nodesource](https://github.com/nodesource/distributions)